src/dx/access/permissions.rs

```diff
@@ -0,0 +1,457 @@
+        self.bits |= UPDATE;
+        self
+    }
+
+    fn delete(mut self) -> Self {
+        self.bits |= DELETE;
+        self
+    }
+}
+
+/// Grant permission to the `channel`.
+///
+/// Create a `channel` permission information object that can be used to specify
+/// required permissions.
+///
+/// # Example
+/// ```rust
+/// # use pubnub::dx::access::permissions::{self, Channel, Permission, ChannelPermission};
+/// #
+/// let channel_permission = permissions::channel("my-channel").get().manage();
+/// # assert_eq!(channel_permission.value(), &0b0010_0100);
+/// ```
+pub fn channel<N>(name: N) -> Box<Channel>
+where
+    N: Into<String>,
+{
+    Box::new(Channel {
+        name: name.into(),
+        bits: 0,
+    })
+}
+
+/// Grant permission to the `channel group`.
+///
+/// Create a `channel group` permission information object that can be used to
+/// specify required permissions.
+///
+/// # Example
+/// ```rust
+/// # use pubnub::dx::access::permissions::{self, Permission, ChannelGroup, ChannelGroupPermission};
+/// #
+/// let channel_group_permission = permissions::channel_group("my-channel-group")
+///     .read();
+/// # assert_eq!(channel_group_permission.value(), &0b0000_0001);
+/// ```
+pub fn channel_group<N>(name: N) -> Box<ChannelGroup>
+where
+    N: Into<String>,
+{
+    Box::new(ChannelGroup {
+        name: name.into(),
+        bits: 0,
+    })
+}
+
+/// Grant permission to the `userId`.
+///
+/// Create a `userId` permission information object that can be used to specify
+/// required permissions.
+///
+/// # Example
+/// ```rust
+/// # use pubnub::dx::access::permissions::{self, Permission, UserId, UserIdPermission};
+/// #
+/// let channel_group_permission = permissions::user_id("my-user-id")
+///     .update();
+/// # assert_eq!(channel_group_permission.value(), &0b0100_0000);
+/// ```
+pub fn user_id<I>(id: I) -> Box<UserId>
+where
+    I: Into<String>,
+{
+    Box::new(UserId {
+        id: id.into(),
+        bits: 0,
+    })
+}
+
+#[cfg(test)]
+mod it_should {
+    use super::*;
+
+    #[test]
+    fn set_proper_read_permission_bits() {
+        assert_eq!(channel("test").read().value(), &0b0000_0001);
+    }
+
+    #[test]
+    fn set_proper_write_permission_bits() {
+        assert_eq!(channel("test").write().value(), &0b0000_0010);
+    }
+
+    #[test]
+    fn set_proper_manage_permission_bits() {
+        assert_eq!(channel("test").manage().value(), &0b0000_0100);
+    }
+
+    #[test]
+    fn set_proper_delete_permission_bits() {
+        assert_eq!(channel("test").delete().value(), &0b0000_1000);
+    }
+
+    #[test]
+    fn set_proper_get_permission_bits() {
+        assert_eq!(channel("test").get().value(), &0b0010_0000);
+    }
+
+    #[test]
+    fn set_proper_update_permission_bits() {
+        assert_eq!(channel("test").update().value(), &0b0100_0000);
+    }
+
+    #[test]
+    fn set_proper_join_permission_bits() {
+        assert_eq!(channel("test").join().value(), &0b1000_0000);
+    }
```

Comment:
You can use `test_case` here ;d

Reply:
@Xavrax, but it doesn't look like it can evaluate expressions. I can't use `channel("test").read().value()` as first argument.

Reply:
Are you sure ;o
I though that it can handle expressions. ;/

Reply:
@Xavrax I tried, but it complains that it expected `,`, but found `.` in the chained call.

Reply:
You can do it like this:
```
    #[test_case(channel("test").read().value() => 0b0000_0001)]
    fn set_proper_read_permission_bits(bits: &u8) -> u8 {
        bits.clone() // <- this clone is awful but it is required because our builder returns borrow.
    }
```

Reply:
@Xavrax nothing changes and output as I mentioned:

<img width="620" alt="Screenshot 2023-04-18 at 10 45 39" src="https://user-images.githubusercontent.com/794617/232724086-0cf6a833-469a-40e5-929e-bd10af900a2c.png">

Reply:
@Xavrax nm, it looks like there is some built-in `test_case` which can not, but if I use dependency – it starts evaluating ;)